utils/managed_bytes.hh

```diff
@@ -394,6 +403,35 @@ public:
     void remove_prefix(size_t prefix_len);
 
     friend class managed_bytes_view;
+    friend class managed_bytes_view_fragment_iterator;
+};
+
+class managed_bytes_view_fragment_iterator : public managed_bytes_view_base {
```

Comment:
There is no reason to inherit from the view, this is just confusing. I know you did it to reuse `remove_prefix()` but I think we shouldn't use that anyway.

Reply:
Also the required member types of an iterator are missing.

Reply:
which methods are missing?
There are: `operator*()`, `operator->()`, `operator++()`, `operator++(int)`
and they seem to be sufficient for all purposes in this series :)

Reply:
```c++
        using iterator_category	= std::forward_iterator_tag;
        using value_type = fragment_type;
        using pointer = const fragment_type*;
        using reference = const fragment_type&;
        using difference_type = std::ptrdiff_t;
```
These guys.

Reply:
At least make this inheritance private, code sharing is its only goal anyway.

Reply:
`manged_bytes_view` calls for `it.remove_prefix`, maybe we can friend it and hide it from external users.

Reply:
This violates the single responsibility principle. The iterator is now also tasked with mutating the underlying view, in fact it is also a view itself (partially), while it should only know how to iterate over the container.